SwiftAudio/Classes/QueueManager.swift

```diff
@@ -147,7 +163,8 @@ class QueueManager<T> {
             throw APError.QueueError.invalidIndex(index: toIndex, message: "The toIndex has to be positive and smaller than the count of current items (\(items.count)).")
         }
         
-        _items.insert(_items.remove(at: fromIndex), at: toIndex)
+        let item = try! removeItem(at: fromIndex)
```

Comment:
I'm using the internal methods because they already account for cases where we'd also have to update the `currentIndex` (i.e) if you remove items that already played.. since your queue would decrease you need to update `currentIndex`.

Reply:
Nice 👍 The "!" should probably be removed from the `try` expressions so the function throws the errors from removeItem(at:) and addItems(items:at:)

Reply:
it's not necessary because we already are making sure the index is valid.. which is the only reason why this method might throw.